daft/session.py

```diff
@@ -478,6 +483,17 @@ def write_table(
 
         self._session.get_table(identifier._ident).write(df, mode=mode, **options)
 
+    ###
+    # functions
+    ###
+    def attach_function(self, func: UDF, alias: str | None = None):
+        """Attaches a Python function as a UDF in the current session."""
+        self._session.attach_function(func, alias)
+
+    def detach_function(self, name: str):
+        """Detaches a Python function as a UDF in the current session."""
+        self._session.detach_function(name)
```

Comment:
`alias` over name for consistency with other attach/detach methods. This one could go either way